samples/data-sealing/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+cmake_minimum_required(VERSION 3.10)
+
+project("Data Sealing Sample" LANGUAGES C CXX)
+
+find_package(Threads REQUIRED)
+
+find_package(OpenEnclave CONFIG REQUIRED)
+add_executable(openenclave::oeedger8r IMPORTED)
+set_target_properties(openenclave::oeedger8r PROPERTIES IMPORTED_LOCATION ${OE_BINDIR}/oeedger8r)
+
+add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/common/datasealing_t.c
+  COMMAND openenclave::oeedger8r --trusted ${CMAKE_SOURCE_DIR}/datasealing.edl
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/common)
+
+# TODO: Move to subdirectories after refactoring
+
+# Enclave 1
+add_executable(enclave_a_v1
+  enc1/ecalls.cpp
+  common/dispatcher.cpp
+  common/keys.cpp
+  common/datasealing_t.c)
+
+target_link_libraries(enclave_a_v1 openenclave::oeenclave openenclave::oelibcxx)
+
+add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/enc1/private.pem ${CMAKE_SOURCE_DIR}/enc1/public.pem
+  COMMAND openssl genrsa -out private.pem -3 3072
+  COMMAND openssl rsa -in private.pem -pubout -out public.pem
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/enc1)
+
+add_custom_command(OUTPUT enclave_a_v1.signed
+  DEPENDS ${CMAKE_SOURCE_DIR}/enc1/private.pem
+  COMMAND openenclave::oesign sign $<TARGET_FILE:enclave_a_v1> ${CMAKE_SOURCE_DIR}/enc1/data-sealing.conf ${CMAKE_SOURCE_DIR}/enc1/private.pem
+  WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+
+# Enclave 2
+add_executable(enclave_a_v2
+  enc2/ecalls.cpp
+  common/dispatcher.cpp
+  common/keys.cpp
+  common/datasealing_t.c)
+
+target_link_libraries(enclave_a_v2 openenclave::oeenclave openenclave::oelibcxx)
+
+add_custom_command(OUTPUT enclave_a_v2.signed
+  DEPENDS ${CMAKE_SOURCE_DIR}/enc1/private.pem
+  COMMAND openenclave::oesign sign $<TARGET_FILE:enclave_a_v2> ${CMAKE_SOURCE_DIR}/enc2/data-sealing.conf ${CMAKE_SOURCE_DIR}/enc1/private.pem
+  WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+
+# Enclave 3
+add_executable(enclave_b
+  enc3/ecalls.cpp
+  common/dispatcher.cpp
+  common/keys.cpp
+  common/datasealing_t.c)
+
+target_link_libraries(enclave_b openenclave::oeenclave openenclave::oelibcxx)
+
+add_custom_command(OUTPUT ${CMAKE_SOURCE_DIR}/enc3/private.pem ${CMAKE_SOURCE_DIR}/enc3/public.pem
+  COMMAND openssl genrsa -out private.pem -3 3072
+  COMMAND openssl rsa -in private.pem -pubout -out public.pem
+  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/enc3)
+
+add_custom_command(OUTPUT enclave_b.signed
+  DEPENDS ${CMAKE_SOURCE_DIR}/enc3/private.pem
+  COMMAND openenclave::oesign sign $<TARGET_FILE:enclave_b> ${CMAKE_SOURCE_DIR}/enc3/data-sealing.conf ${CMAKE_SOURCE_DIR}/enc3/private.pem
+  WORKING_DIRECTORY ${CMAKE_BINARY_DIR})
+
+# Host
+add_subdirectory(host)
+
+add_custom_target(run
```

Comment:
For my understanding, is this a common pattern in CMake? I assume it was done to mirror the Makefiles, but if there's a more idiomatic way to do this in CMake (e.g. ctest?) then we should use it as it would be more intuitive for its target audience.

Reply:
So much more idiomatic ways, but this pass was to mirror GNU Make 😋 If we can remove the Makefiles, I can write this way better. My proposal to @soccerGB was to have two sets of samples, `samples/pkg-config` and `samples/cmake` so we don't have to do weird things.

Reply:
Can we track this as an unresolved issue? I don't believe we should remove the make samples, but they should still be convergent in their design and we should feel free to restructure the build structure as appopriate (e.g. as has been done with the common code, but not the test signing of individual enclaves and the generation of test keys)

Reply:
This actually worked out rather well, as we want a custom target to run the sample, since the actual test is located a context up in the 'upstream' project. So the only question is the name, and I see no compelling reason to change it. If this were standalone, then maybe `add_test(foo)` but it'd be more complicated, and not worth it IMO.